lumeer-core/src/main/java/io/lumeer/core/facade/AbstractFacade.java

```diff
@@ -48,11 +59,20 @@
       return resource;
    }
 
-   public void keepStoredPermissions(final Resource resource, final Permissions storedPermissions) {
+   protected void keepStoredPermissions(final Resource resource, final Permissions storedPermissions) {
       Set<Permission> userPermissions = storedPermissions.getUserPermissions();
       resource.getPermissions().updateUserPermissions(userPermissions.toArray(new Permission[0]));
 
       Set<Permission> groupPermissions = storedPermissions.getGroupPermissions();
       resource.getPermissions().updateGroupPermissions(groupPermissions.toArray(new Permission[0]));
    }
+
+   protected SearchQuery createSearchQuery(Pagination pagination) {
+      String user = authenticatedUser.getCurrentUsername();
+      Set<String> groups = userCache.getUser(user).getGroups();
+
```

Comment:
Pagination is here for future use, am I right?

Reply:
No, it should have already been used. But I haven't written any tests for the pagination, that's why I haven't discovered this problem. Nice catch! Thanks.